src/v2/mod.rs

```diff
@@ -755,6 +917,122 @@ impl Client {
+        let peer_connection = generated::IpSocketAddress {
+            ip:   Some(generated::IpAddress {
+                value: addr.ip().to_string(),
+            }),
+            port: Some(generated::Port {
+                value: addr.port() as u32,
+            }),
+        };
+        self.client.peer_disconnect(peer_connection).await?;
+        Ok(())
+    }
+
+    /// Get a vector of the banned peers.
+    pub async fn get_banned_peers(
+        &mut self,
+    ) -> endpoints::RPCResult<Vec<super::types::network::BannedPeer>> {
+        Ok(self
+            .client
+            .get_banned_peers(generated::Empty::default())
+            .await?
+            .into_inner()
+            .peers
+            .into_iter()
+            .map(super::types::network::BannedPeer::try_from)
+            .collect::<anyhow::Result<Vec<super::types::network::BannedPeer>>>()?)
+    }
+
+    /// Ban a peer
+    /// Returns whether the peer was banned or not.
+    pub async fn ban_peer(
+        &mut self,
+        peer_to_ban: super::types::network::PeerToBan,
+    ) -> endpoints::RPCResult<()> {
+        self.client.ban_peer(peer_to_ban).await?;
+        Ok(())
+    }
+
+    /// Unban a peer
+    /// Returns whether the peer was unbanned or not.
+    pub async fn unban_peer(
+        &mut self,
+        banned_peer: &super::types::network::BannedPeer,
+    ) -> endpoints::RPCResult<()> {
+        self.client.unban_peer(banned_peer).await?;
+        Ok(())
+    }
+
+    /// Start a network dump if the feature is enabled on the node.
+    /// Return true if a network dump has been initiated.
+    ///
+    /// * file - The file to write to.
+    /// * raw - Whether raw packets should be included in the dump or not.
+    ///
+    /// Note. If the feature 'network_dump' is not enabled on the node then this
+    /// will return a 'Precondition failed' error.
+    pub async fn dump_start(&mut self, file: String, raw: bool) -> endpoints::RPCResult<()> {
+        self.client
+            .dump_start(generated::DumpRequest { file, raw })
+            .await?;
+        Ok(())
+    }
+
+    /// Stop an ongoing network dump.
+    /// Return true if it was successfully stopped otherwise false.
+    ///
+    /// Note. If the feature 'network_dump' is not enabled on the node then this
+    /// will return a 'Precondition failed' error.
+    pub async fn dump_stop(&mut self) -> endpoints::RPCResult<()> {
+        self.client.dump_stop(generated::Empty::default()).await?;
+        Ok(())
+    }
+
+    /// Retrieve information about the peers that the node is connected to.
+    pub async fn get_peers_info(&mut self) -> endpoints::RPCResult<types::network::PeersInfo> {
+        let response = self
+            .client
+            .get_peers_info(generated::Empty::default())
+            .await?;
+        let peers_info = types::network::PeersInfo::try_from(response.into_inner())?;
+        Ok(peers_info)
+    }
+
+    /// Retrieve [types::NodeInfo] from the node. 
```

Comment:
This is not super useful. Perhaps elaborate what this is compared to `get_consensus_info`, so users may have an easier time using the right one.